spec/requests/users_controller_spec.rb

```diff
@@ -834,6 +834,42 @@ def enabled?
         end
       end
 
+      context "when normalize_emails is enabled" do
+        let (:email) {
+          "jane+100@gmail.com"
+        }
+        let (:dupe_email) {
+          "jane+191@gmail.com"
+        }
+        let! (:user) {
+          Fabricate(:user, email: email, password: "strongpassword")
+        }
+
+        before do
+          SiteSetting.hide_email_address_taken = true
+          SiteSetting.normalize_emails = true
+        end
+
+        it "sends an email to normalized email owner when hide_email_address_taken is enabled" do
+          expect do
+            post "/u.json",
+                 params: {
+                   name: "Jane Doe",
+                   username: "janedoe9999",
+                   password: "strongpassword",
+                   email: dupe_email,
+                 }
+          end.to_not change { User.count }
+
+          expect(response.status).to eq(200)
+          expect(session["user_created_message"]).to be_present
+
+          # expecting a warning emails
+          expect(Jobs::CriticalUserEmail.jobs.size).to eq(1)
+          expect(Jobs::CriticalUserEmail.jobs.first["args"].first["type"]).to eq("account_exists")
```

Comment:
We have helpers that can simplify the assertions here which you may want to use.

https://github.com/discourse/discourse/blob/80dcaf1e9842deb5afb4652c8efb57a6aa973c99/spec/support/sidekiq_helpers.rb